### DISCUSSION TITLE
Failed to ensure HPA is correctly created for ScaledObject

### DISCUSSION THREAD

**Discussion by ndoddi-conga** [kedacore/keda#1338] [Q&A / Need Help] [answered]
I Installed KEDA Latest version and deployed ScaledObject in my cluster and followed all instructions to enable KEDA with KAFKA scaler but once i deployed. HPA is not enabled and i am facing this issue when i described my ScaledObject

here is the error:
Status:
  Conditions:
    Message:               **Failed to ensure HPA is correctly created for ScaledObject**
    Reason:                ScaledObjectCheckFailed
    Status:                False
    Type:                  Ready
    Message:               ScaledObject check failed
    Reason:                UnkownState
    Status:                Unknown
    Type:                  Active
  Original Replica Count:  1
  Scale Target GVKR:
    Group:            apps
    Kind:             Deployment
    Resource:         deployments
    Version:          v1
  Scale Target Kind:  apps/v1.Deployment
Events:               <none>

I am not sure where i am going wrong. I removed my kafka details from the triggers

here is my **ScaledObject.yaml** File

apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: kafka-keda-scaledobject
  namespace: default
spec:
  scaleTargetRef:
    name: kafka-consumer # Required Name of the deployment we want to scale.
    kind: Deployment
  pollingInterval: 5
  minReplicaCount: 0   #Optional Default 0
  maxReplicaCount: 10  #Optional Default 100
  advanced:                                          # Optional. Section to specify advanced options
    restoreToOriginalReplicaCount: true        # Optional. Default: false
    horizontalPodAutoscalerConfig:                   # Optional. Section to specify HPA related options
      behavior:                                      # Optional. Use to modify HPA's scaling behavior
        scaleDown:
          stabilizationWindowSeconds: 300
          policies:
          - type: Percent
            value: 100
            periodSeconds: 15
  triggers:
  - type: kafka
    metadata:
      # Required
      brokerList: <Removed> # Kafka bootstrap server host and port
      consumerGroup: <>  # Make sure that this consumer group name is the same one as the one that is consuming topics
      topic: <>
      lagThreshold: "10" # Optional. How much the stream is lagging on the current consumer group

**Comment by zroubalik** [comment, marked as answer]
could you please post logs from the KEDA operator?

**Reply by ndoddi-conga** [reply]
Here are the logs from KEDA operator

{"level":"error","ts":1605723933.6586707,"logger":"controller_scaledobject","msg":"Failed to create new HPA resource","Request.Namespace":"default","Request.Name":"kafka-keda-scaledobject","HPA.Namespace":"default","HPA.Name":"keda-hpa-kafka-consumer","error":"error getting scaler for trigger #0: error creating kafka client: kafka: client has run out of available brokers to talk to (Is your cluster reachable?)","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/zroubali/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/kedacore/keda/pkg/controller/scaledobject.(*ReconcileScaledObject).reconcileDeploymentType\n\tkeda/pkg/controller/scaledobject/scaledobject_controller.go:236\ngithub.com/kedacore/keda/pkg/controller/scaledobject.(*ReconcileScaledObject).Reconcile\n\tkeda/pkg/controller/scaledobject/scaledobject_controller.go:155\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/zroubali/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:216\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/zroubali/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:192\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/Users/zroubali/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:171\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/Users/zroubali/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/Users/zroubali/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/Users/zroubali/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:88"}



Based on this i found it like i need to provide my KAFKA credentials and Cert details but i can able to provide username and password but not sure what will fit here 

apiVersion: v1
kind: Secret
metadata:
  name: keda-kafka-secrets
  namespace: default
data:
  authMode: "sasl_plaintext"
  username: ""
  password: ""
  **ca: <your ca>
  cert: <your cert>
  key: <your key>**

like i have kafka cert in my hand but not sure what details will fit under this highlighted one if possible can you share me some example like what details need to fit here then i can add w.r.t my cert

**Reply by zroubalik** [reply]
you can see example in the docs https://keda.sh/docs/2.0/scalers/apache-kafka/

**Reply by ndoddi-conga** [reply]
thank you zroubalik for sending the link but from it there is no much details mentioned about SSL cert like i do have server cert lets say kafka.pem file but how i can fit this in these values since i am not aware what details need to provide here.
![image](https://user-images.githubusercontent.com/63255798/99705476-a04abb80-2a4e-11eb-91d7-d2c855a15580.png)

like 

**ca: <your ca>
cert: <your cert>
key: <your key>**

i just have a cert kafka.pem cert and i mount this every time to a consumer and producer pod to connect to kafka but since ScaledObject kind is different and i am not aware how i can fit the cert in the above details ?

**Reply by zroubalik** [reply]
I think you can convert `.pem` to `.cert` and `.key` and reference it here (eg. https://stackoverflow.com/questions/13732826/convert-pem-to-crt-and-key)

**Reply by tomkerkhove** [reply]
@ndoddi-conga Does this answer your question?

**Reply by ndoddi-conga** [reply]
Yes, Thanks once i provide all the cert's encoded keys KEDA can able to connect to KAFKA brokers and scale it up

**Reply by tomkerkhove** [reply]
Great to hear!
